src/high-level/types/complex-double-float.lisp

```diff
@@ -0,0 +1,255 @@
+;;;; complex-double-float.lisp
+;;;;
+;;;; Author: Cole Scott
+
+(in-package #:magicl)
+
+;; TODO: Is there a better way to do this?
+(defconstant +double-comparison-threshold-loose+  1d-5)
+(defconstant +double-comparison-threshold-strict+ 5d-11)
+
+(deftensor tensor/complex-double-float (complex double-float))
+
+(defmatrix matrix/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defvector vector/complex-double-float (complex double-float) tensor/complex-double-float)
+
+(defcompatible
+    (lambda (tensor)
+      (case (order tensor)
+        (1 '(vector/complex-double-float
+             tensor/complex-double-float))
+        (2 '(matrix/complex-double-float
+             tensor/complex-double-float))
+        (t '(tensor/complex-double-float))))
+  tensor/complex-double-float
+  matrix/complex-double-float
+  vector/complex-double-float)
+
+(defmethod dot ((vector1 vector/complex-double-float) (vector2 vector/complex-double-float))
+  (assert (cl:= (size vector1) (size vector2))
```

Comment:
Should these assertions and type-checks be wrapped in policy-conds? If yes, make a TODO/issue.